### ISSUE TITLE
Leaks memory, until it crashes

### ISSUE THREAD

**Issue by Pictor13** [sienori/Tab-Session-Manager#941]
## Short description
The extension is leaking memory as hell.

## Expected result
Previous behaviour: low memory footprint & no crashes/reloads.

## Actual result

At loading time the extension occupied around 150/200MB of RAM; in 10 minutes from loading it reached 2.2GB. Some 5 more minutes passed and now it's 3.2GB (but I was on the same tab, writing this text).
And after few other minutes it crashes.

This repeats ad infinitum.

Here I pasted the log, from the extension reload till the now: https://pastebin.com/6dsiq2Rs \
It doesn't of course contain the crash log, since the background.html is also getting closed.

## Platform information
* Platform (OS): macOS Monterey (was broken also on Mojave)
* Version of browser: Chrome 96.0.4664.110 (Official Build) (x86_64)
* Version of Tab Session Manager: 6.11.1

## Additional context

Happens since some months, sorry for not opening a ticket earlier. Before I didn't have any issue and the extension was working perfectly.\
I didn't install other extensions since then.

This, with about 300 tabs open. I sometime get >1000; but that was never a problem for TabSessionManager, it was working flawlessly; until few months ago.

I had to stop using the extension at all because it hogs resources and then crashes repeatedly.

----
I also got this error in the console:

> *Refused to load the font 'https://github.com/sienori/res/Soleil.woff2' because it violates the following Content Security Policy directive: "font-src github.githubassets.com".* - `/sienori/Tab-Session-Manager:1`

but I am not sure if that's related.

**Comment by DrSpaceCake** [comment]
Not sure if it's the same problem that I'm experiencing, I don't really know coding;
However, I can tell you what I've observed:

When starting FF: It takes about one minute, and it'll have grabbed any and all available RAM, usually around 10gb, then it proceeds to fill up the commit memory with around 20-30gb of idk what... It won't hog too much CPU %, I guess most of the CPU usage, is used to write the absurd amount of data into commit. Finally, even while using this absurd amount of resources, it takes forever to load up, when I actually want to use it to save/load a session. 
After a few minutes of having filled up the commit memory to whatever it is that makes it "happy," it'll decrease it's ram usage, until I start using the browser actively, (opening/closing tabs/windows, or just clicking links/watching youtube, etc.) and it'll start eating RAM again. 

Other effects it had:
When loading several tabs, or browsing through links fast, it tended to just freeze the browser's ability to ... browse; this was particularly bad, when I had several windows open at the same time.


I'm not certain when this started, I figured it had something to do with the 3+k tabs I had open... Or so I thought, until I tried closing all of them, except a few > played around with task manager > searched online > tried disabling extensions 1 by 1.


Platform info:
Extension version: 6.11.0
Browser: Firefox 98.0.2 (64-bit)
OS: Win 10 pro


Thanks in advance.

**Comment by hkben** [comment]
I use this extension on both Firefox and Chrome. I got same issue on Firefox only, and completely fine on Chrome. I did some memory checking on Firefox recently.

The problem is the log (with session array) is piling up in console, and it eats up memory very quickly. Simply clear the console releases the memory.

Of course the best solution is disable Debug Mode. I turned it off and kept Firefox running for a day. It completely fine, only ~700mb memory used on Firefox's extension process and no crashes.
